File: src/main/java/net/rk4z/mcef/glfw/MCEFGlfwCursorHelper.java
/*
 *     MCEF (Minecraft Chromium Embedded Framework)
 *     Copyright (C) 2024 Ruxy
 *
 *     This library is free software; you can redistribute it and/or
 *     modify it under the terms of the GNU Lesser General Public
 *     License as published by the Free Software Foundation; either
 *     version 2.1 of the License, or (at your option) any later version.
 *
 *     This library is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 *     Lesser General Public License for more details.
 *
 *     You should have received a copy of the GNU Lesser General Public
 *     License along with this library; if not, write to the Free Software
 *     Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301
 *     USA
 */

package net.rk4z.mcef.glfw;

import org.cef.misc.CefCursorType;
import org.lwjgl.glfw.GLFW;

import java.util.HashMap;

public class MCEFGlfwCursorHelper {

    private static final HashMap<CefCursorType, Long> CEF_TO_GLFW_CURSORS = new HashMap<>();

    /**
     * Helper method to get a GLFW cursor handle for the given {@link CefCursorType} cursor type
     */
    public static long getGLFWCursorHandle(CefCursorType cursorType) {
        if (CEF_TO_GLFW_CURSORS.containsKey(cursorType)) {
            return CEF_TO_GLFW_CURSORS.get(cursorType);
        }

        var glfwCursorHandle = GLFW.glfwCreateStandardCursor(cursorType.glfwId);
        CEF_TO_GLFW_CURSORS.put(cursorType, glfwCursorHandle);
        return glfwCursorHandle;
    }

}

File: src/main/java/net/rk4z/mcef/glfw/MCEFGlfwCursorHelper.kt
package net.rk4z.mcef.glfw

import org.cef.misc.CefCursorType
import org.lwjgl.glfw.GLFW

object MCEFGlfwCursorHelper {
    private val CEF_TO_GLFW_CURSORS = HashMap<CefCursorType, Long>()

    /**
     * Helper method to get a GLFW cursor handle for the given [CefCursorType] cursor type
     */
    fun getGLFWCursorHandle(cursorType: CefCursorType): Long {
        if (CEF_TO_GLFW_CURSORS.containsKey(cursorType)) {
            return CEF_TO_GLFW_CURSORS[cursorType]!!
        }

        val glfwCursorHandle = GLFW.glfwCreateStandardCursor(cursorType.glfwId)
        CEF_TO_GLFW_CURSORS[cursorType] = glfwCursorHandle
        return glfwCursorHandle
    }
}
